enmanuel-submission

emtes opened the pull request.

ROgbonna1 commented on linkedlist.py:
```diff
@@ -0,0 +1,118 @@
+            self.head = self.head.next
+        return c
+
+    def __iter__(self):
```
👍

ROgbonna1 commented on linkedlist.py:
```diff
@@ -0,0 +1,118 @@
+            yield self.head
+            self.head = self.head.next
+
+    def __contains__(self, data):
```
👍

ROgbonna1 commented on linkedlist.py:
```diff
@@ -0,0 +1,118 @@
+            self.head = self.head.next
+        return False
+
+    def append(self, data):
```
Take another look at this. We can append to the head but then we want to ensure that the new `head` is the newly created node.

ROgbonna1 commented on linkedlist.py:
```diff
@@ -0,0 +1,118 @@
+    return False
+
+# Question 3: Reverse a Linked List
+def reverse(head: Node) -> Node:
```
👍

ROgbonna1 commented on linkedlist.py:
```diff
@@ -0,0 +1,118 @@
+                current.next = None
+
+# Question 2: Cycle Check
+def is_cyclic(head: Node) -> bool:
```
👍

ROgbonna1 commented on linkedlist.py:
```diff
@@ -0,0 +1,118 @@
+    return head
+
+# Question 5: Remove duplicates
+def remove_dups(head: Node) -> Node:
```
Well done

ROgbonna1 commented on linkedlist.py:
```diff
@@ -0,0 +1,118 @@
+    return pre
+
+# Question 4: Merge Two Lists
+def merge_two_list(l1: Node, l2: Node) -> Node:
```
Awesome!

emtes commented on linkedlist.py:
```diff
@@ -0,0 +1,118 @@
+        self.value = data
+        self.next = None
+
+    def __repr__(self):
```
If I add `.value` it actually throws an `AttributeError` if the node's `.next` = `None`. Like it is now prints `None`. E.g. `4 -> None`

emtes commented:
Refactored everything to have `current` variable to make it more readable, changed the `repr` magic method, old one is actually really unpleasant to read when testing list. Reviewed and refactored merging lists.
